### DISCUSSION TITLE
Reading Production not just Consumption

### DISCUSSION THREAD

**Discussion by heyhewmike** [bemasher/rtlamr#327] [General]
I have solar and I have 2 meters installed on the side of my house. 1. Is for tracking Solar Production. 2. Is for Grid Consumption & Grid Sell Back.

I am presently only seeing consumption from the meters and not the production in the output in my terminal. My meters are OpenWay by Itron Model AMI7 FCC ID SK9AMI7 using SCM+.

Is there anyway that I can see the other data such as the Grid Return (Sell Back) & the tracking of the Production meter?

From Grid (Consumption) is displayed as a '4' on the meter.

To Grid or Solar Production (Depending on meter position in the Electrical system in my house) is displayed as a '10' on the meter.
Production (on Production meter) is between the solar panels and everything.
To Grid (On Billing Meter) = Production - What the house has used.

Is there a way that I am missing to see this data or is this not part of the data being provided?

I am willing to perform any needed testing with my meters.

I have attached a PDF from my grid provider on how to read Consumption vs Production on the meters.
[eversource-net-metering-faq .pdf](https://github.com/user-attachments/files/17132141/eversource-net-metering-faq.pdf)

**Comment by NordicChinaBusinessHub** [comment]
You might need to contact your utility provider for clarification or consider using energy management software; legal assistance for such matters is something we often deal with at Nordic China Business Hub.

**Comment by heyhewmike** [comment]
> You might need to contact your utility provider for clarification or consider using energy management software; legal assistance for such matters is something we often deal with at Nordic China Business Hub.

I do not know enough about Go or this tool but I am making a few inferences. If I am able to detect my consumption, Shown as a '04'on the LCD , on SCM+ at ProtocolID:0x1E and EndpointType:0x6E as Consumption then would I not be able to detect my production, Shown as a '10'on the LCD, at the same ProtocolID with a different EndpointType?

What energy management software do you feel I would need or what clarification would I need from my utility provider?
I do not feel the standard support rep I would be able to speak with would understand or be able to provide the hex value needed to identify Production picked up OTA or provide me assistance with this tool.

**Comment by kidhasmoxy** [comment]
@heyhewmike - did you ever get an answer here? It looks like the protocol for scm+ will only ever grab a consumption message, so the meter would have to broadcast a different ID to pick it up. I've never seen a broadcast for those, though. Wondering if there's a different message format they're using.
